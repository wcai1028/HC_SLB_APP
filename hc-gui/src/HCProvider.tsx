import React from 'react'
import { Store } from 'redux'
import { A10Provider } from 'a10-gui-framework'

export interface IHCPRoviderProps {
  children: JSX.Element
  store?: Store<any>
  initState?: IObject
}

import appReducers from './redux/reducers'
import config from './settings/config'
const middlewares: any[] = []

const HCProvider = (props: IHCPRoviderProps) => {
  const { children, store, initState } = props
  const options: any = {
    initState,
    reducers: appReducers,
    middlewares,
    CONFIG: config,
    children,
  }
  if (store) {
    options.store = store
  }
  return <A10Provider {...options} />
}
export default HCProvider
