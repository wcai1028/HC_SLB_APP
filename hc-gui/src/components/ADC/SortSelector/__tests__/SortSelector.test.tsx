import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'
import * as moxios from 'moxios'
import HCProvider from 'src/HCProvider'

const renderTestComponent = () => {
  const SortSelector = require('../index').default
  return ReactTestRenderer.create(
    <HCProvider>
      <SortSelector />
    </HCProvider>,
  )
}
const renderTestComponentWithProps = () => {
  const SortSelector = require('../index').default
  return ReactTestRenderer.create(
    <HCProvider>
      <SortSelector
        defaultSelected="Time"
        name="alert-sort-selector"
        sortTitle="Sort By"
        sortList={['Time', 'Severity', 'Object Type', 'Name']}
      />
    </HCProvider>,
  )
}

const getComponentInstance = (testRenderer: any) => {
  const testInstance = testRenderer.root
  const testComponent = testInstance.find(
    (node: any) => node.type.name === 'SortSelector',
  )
  return testComponent.instance
}
beforeEach(() => {
  moxios.install()
})
afterEach(() => {
  moxios.uninstall()
})

describe('test component: SortSelector', () => {
  test('should render properly', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent()
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot('without Props')
      done()
    })
  })
  test('should render properly with Props', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponentWithProps()
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot('with Props')
      done()
    })
  })
})

describe('test function literally', () => {
  test('componentWillReceiveProps: should render properly', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponentWithProps()
    const instance = getComponentInstance(testComponent)
    instance.componentWillReceiveProps({ defaultSelected: 'Time' })
    setImmediate(() => {
      expect(instance.state.selectedValue).toBe('Time')
      done()
    })
  })
  test('handleClick: should render properly', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponentWithProps()
    const instance = getComponentInstance(testComponent)
    instance.handleClick({ key: 'Severity' })
    setImmediate(() => {
      expect(instance.state.selectedValue).toBe('Severity')
      done()
    })
  })
})
