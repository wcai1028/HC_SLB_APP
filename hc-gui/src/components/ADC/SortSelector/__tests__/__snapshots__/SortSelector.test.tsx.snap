// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: SortSelector should render properly with Props: with Props 1`] = `
<div
  className="sortSelector"
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <span
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="alert-sort-selector sortSelectorName"
    >
      Time
    </span>
    <span
      app="global"
      className="a10-widget-appicon"
      style={
        Object {
          "height": "12px",
          "lineHeight": "20px",
          "marginLeft": "4px",
          "width": "12px",
        }
      }
      type="dropdown"
    >
      <img
        src={[MockFunction]}
        style={
          Object {
            "height": "12px",
            "width": "12px",
          }
        }
      />
    </span>
  </span>
</div>
`;

exports[`test component: SortSelector should render properly: without Props 1`] = `
<div
  className="sortSelector"
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <span
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="undefined sortSelectorName"
    />
    <span
      app="global"
      className="a10-widget-appicon"
      style={
        Object {
          "height": "12px",
          "lineHeight": "20px",
          "marginLeft": "4px",
          "width": "12px",
        }
      }
      type="dropdown"
    >
      <img
        src={[MockFunction]}
        style={
          Object {
            "height": "12px",
            "width": "12px",
          }
        }
      />
    </span>
  </span>
</div>
`;
